docs/guide/supervisor-programming.md

```diff
@@ -111,12 +229,177 @@ int main() {
     }
   }
 
-my_exit:
-  wb_robot_cleanup();
+  return my_exit();
+}
+```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <webots/Supervisor.hpp>
+
+using namespace webots;
 
+int cleanUp(Supervisor *supervisor) {
+  delete supervisor;
   return 0;
 }
+
+int main(int argc, char **argv) {
+  Supervisor *supervisor = new Supervisor();
+
+  // get handle to robot's translation field
+  int timeStep = (int)supervisor->getBasicTimeStep();
+  Node *robotNode = supervisor->getFromDef("MY_ROBOT");
+  Field *translation = robotNode->getField("translation");
+
+  double a, b, t;
+  for (a = 0.0; a < 5.0; a += 0.2) {
+    for (b = 0.0; b < 10.0; b += 0.3) {
+      // evaluate robot during 60 seconds (simulation time)
+      for (t = 0.0; t < 60.0; t += timeStep / 1000.0) {
+
+        // perform robot control according to a, b
+        // (and possibly t) parameters.
+
+        // controller termination
+        if (supervisor->step(timeStep) == -1)
+          return cleanUp(supervisor);
+      }
+
+      // compute travelled distance
+      const double *pos = translation->getSFVec3f();
+      double dist = sqrt(pos[0] * pos[0] + pos[2] * pos[2]);
+      std::cout << "a=" << a << " b=" << b << " dist=" << dist << std::endl;
+
+      // reset robot position and physics
+      const double INITIAL[3] = {0, 0.5, 0};
+      translation->setSFVec3f(INITIAL);
+      robotNode->resetPhysics();
+    }
+  }
+  // end of tests
+  return cleanUp(supervisor);
+}
+```
+%tab-end
+
+%tab "Python"
+```python
+from math import sqrt
+from controller import Supervisor
+
+supervisor = Supervisor()
+
+timestep = int(supervisor.getBasicTimeStep())
+robot_node = supervisor.getFromDef("MY_ROBOT")
+translation_field = robot_node.getField("translation")
+
+for a in range(0, 25):
+    for b in range(0, 33):
```

Comment:
Can't we use the same values (integers is probably simpler) in all languages. Also the comments should be the same and at the same place across different languages.